test/config/targetstatetest.go

```diff
@@ -29,12 +28,21 @@ func (s *TestSuite) TestDeviceState(t *testing.T) {
 	defer gnmi.DeleteSimulator(t, simulator)
 
 	assert.NotNil(t, simulator)
-	found := gnmi.WaitForTarget(t, func(d *device.Device, eventType topo.EventType) bool {
-		return len(d.Protocols) > 0 &&
-			d.Protocols[0].Protocol == topo.Protocol_GNMI &&
-			d.Protocols[0].ConnectivityState == topo.ConnectivityState_REACHABLE &&
-			d.Protocols[0].ChannelState == topo.ChannelState_CONNECTED &&
-			d.Protocols[0].ServiceState == topo.ServiceState_AVAILABLE
+	found := gnmi.WaitForTarget(t, func(d *topo.Object, eventType topo.EventType) bool {
+		protocols := &topo.Protocols{}
+		err := d.GetAspect(protocols)
+		assert.NoError(t, err)
+		var protocolStates []*topo.ProtocolState
+		if err != nil {
+			protocolStates = nil
+		} else {
+			protocolStates = protocols.State
+		}
+		return len(protocolStates) > 0 &&
+			protocolStates[0].Protocol == topo.Protocol_GNMI &&
+			protocolStates[0].ConnectivityState == topo.ConnectivityState_REACHABLE &&
```

Comment:
We should not check these states anymore. The control relation defines connectivity. The entity can be in topo but not connected. I guess we should remove protocolState all together in tests.